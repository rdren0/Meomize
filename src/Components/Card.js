import React, { Component } from 'react';

class Card extends Component {
  constructor(){
    super();
    this.state = {
      showAnswer: false
    }
  }


  answerVisible(){
    this.setState({showAnswer:!showAnswer})

  }

render() {
    return (
      <div>
       <div className="card-box">
          <h3 className="card-title">
            {this.props.cardTitle}
          </h3>
          <button onClick= {this.answerVisible} className="show-answer">Show Answer</button>
        </div>
        {this.state.showAnswer &&
        <div className="hidden-answer">
            <h2>
            {this.props.cardAnswer}
            </h2>
            <button onClick= {this.answerVisible}>Try Again</button>
            <button className="remove-card">Delete Card</button>
        </div>}
      </div>
    );
  }
}


export default Card;
