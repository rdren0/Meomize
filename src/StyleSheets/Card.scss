
.card-box{
display: grid;
justify-items: center;
grid-template-areas: "title" "button";
grid-template-rows: 1fr 50px;
border-radius: 15px;
width: 300px;
height: 180px;
border: 1px solid gainsboro;
background-color: #79C7C5;
margin: 10px;
padding: 15px;

}

.card-box:hover{
  background-color: #69ADAB;
  -webkit-box-shadow: 0px 10px 21px -5px rgba(153,144,153,0.63);
  -moz-box-shadow: 0px 10px 21px -5px rgba(153,144,153,0.63);
  box-shadow: 0px 10px 21px -5px rgba(153,144,153,0.63);
}

.card-box button:hover{
  background-color: #E1CE7A;
  color: black;
  font-weight: 700;
  cursor: pointer;
}

.answer-box{
border-radius: 15px;
width: 300px;
height: 180px;
border: 1px solid gainsboro;
background-color: #79C7C5;
margin: 10px;
padding: 15px;
}

.hidden-answer{
  display: none;
}

.show-answer{
  grid-area: button;
}

.card-title{
  grid-area: title;
 
}

.remove-card{
  grid-area: remove;
}