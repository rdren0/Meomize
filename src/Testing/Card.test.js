import React from 'react';
import Card from '../Components/Card.js';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
const answerVisible = jest.fn();

const mockCard = {
"id": 2,
"type": "syntax",
"title": "What goes in the parameters of a Reduce?",
"answer": "Acc and Current Value",
"complete": false
}

describe('Card', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Card
                key={mockCard.id} 
                cardType={mockCard.type} 
                cardTitle={mockCard.title} 
                cardAnswer={mockCard.answer} />
    )
  })
  
  it('should match the snapshot with all data passed in' , () =>{
    expect(wrapper).toMatchSnapshot();
  });

  it('should toggle answer', () => {
    wrapper.find('.show-answer').simulate('click');
    expect(wrapper.state('showAnswer')).toEqual(true);
  });
});



