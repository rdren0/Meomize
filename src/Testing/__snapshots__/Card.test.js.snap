// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with all data passed in 1`] = `
<div
  className="card-box"
>
  <h3
    className="card-title"
  >
    What goes in the parameters of a Reduce?
  </h3>
  <button
    className="show-answer"
    onClick={[Function]}
  >
    Show Answer
  </button>
</div>
`;
